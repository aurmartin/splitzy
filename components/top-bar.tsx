import { Text } from "@/components/text";
import { Ionicons } from "@expo/vector-icons";
import React from "react";
import { Pressable, View } from "react-native";
import { useRouter } from "expo-router";

interface TopBarProps {
  title?: string;
  rightActions?: React.ReactNode[] | React.ReactNode;
}

export const TopBar = (props: TopBarProps) => {
  const { title, rightActions } = props;
  const router = useRouter();

  return (
    <View
      style={{
        flexDirection: "row",
        alignItems: "center",
        marginBottom: 16,
      }}
    >
      <Pressable
        style={{ backgroundColor: "white", padding: 8, borderRadius: 24 }}
        onPress={() => router.back()}
        android_ripple={{ color: "hsl(0 0% 90%)" }}
        testID="topbar-back-button"
      >
        <Ionicons name="arrow-back-outline" size={24} color="hsl(0 0% 40%)" />
      </Pressable>

      {title ? (
        <Text type="title" style={{ marginLeft: 16 }}>
          {title}
        </Text>
      ) : null}

      {rightActions ? (
        <View style={{ marginLeft: "auto", flexDirection: "row", gap: 8 }}>
          {rightActions}
        </View>
      ) : null}
    </View>
  );
};
