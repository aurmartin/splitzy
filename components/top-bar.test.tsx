import * as React from "react";
import { render, screen, userEvent } from "@testing-library/react-native";
import { TopBar } from "./top-bar";
import { Text, Pressable } from "react-native";

// Mock expo-router
const mockBack = jest.fn();
jest.mock("expo-router", () => ({
  useRouter: () => ({
    back: mockBack,
  }),
}));

describe("TopBar", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("renders correctly", () => {
    render(<TopBar />);
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it("renders title when provided", () => {
    const title = "Test Title";
    render(<TopBar title={title} />);

    screen.getByText(title);
  });

  it("handles back button press correctly", async () => {
    const user = userEvent.setup();
    render(<TopBar />);

    await user.press(screen.getByTestId("topbar-back-button"));
    expect(mockBack).toHaveBeenCalled();
  });

  it("renders right actions when provided", () => {
    const actionText = "Action";
    const rightActions = [
      <Pressable key="1" testID="right-action">
        <Text>{actionText}</Text>
      </Pressable>,
    ];

    render(<TopBar rightActions={rightActions} />);

    screen.getByTestId("right-action");
    screen.getByText(actionText);
  });
});
