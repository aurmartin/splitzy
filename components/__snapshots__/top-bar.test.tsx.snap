// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "marginBottom": 16,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "white",
        "borderRadius": 24,
        "padding": 8,
      }
    }
    testID="topbar-back-button"
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "hsl(0 0% 40%)",
            "fontSize": 24,
          },
          undefined,
          {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      î¨¬
    </Text>
  </View>
</View>
`;
